### PR TITLE
Queries Aggregation Command

### PR DESCRIPTION
**Pull request by YuriHechter**

### CODE REVIEW

File: cmd/search_log.go [hunk 1]
```diff
@@ -354,10 +364,91 @@ func latencyAggregateFn(cmd *cobra.Command, args []string) {
...
+	RESULTS_FOR_LANGUAGE := 1000
+	gteStr := null.NewString("", false)
+	languages := []string{consts.LANG_HEBREW, consts.LANG_ENGLISH, consts.LANG_RUSSIAN} // Currently only the 3 main languages are supported due to language recognition issues
+	queryCountByLang := map[string]map[string]int{}                                     // Language -> [Search Term -> Count]
+	if lastDays > 0 {
+		gteStr = null.StringFrom(fmt.Sprintf("now-%dd/d", lastDays))
+	}
+	for i := 0; i < SLICES; i++ {
+		s := elastic.NewSliceQuery().Id(i).Max(SLICES)
+		queries, err := logger.GetLattestQueries(s, gteStr, null.NewBool(false, false))
```
Comment (bbfsdev) [inline cmd/search_log.go:388]: What GetLattestQueries does?

Comment (bbfsdev) [reply cmd/search_log.go:388]: How did you test it works well?

File: cmd/search_log.go [hunk 2]
```diff
@@ -354,10 +364,91 @@ func latencyAggregateFn(cmd *cobra.Command, args []string) {
...
+		utils.Must(err)
+		for _, sl := range queries {
+			term := simpleQuery(sl.Query)
+			if term == "" || sl.Error != nil || sl.QueryResult == nil || sl.Query.Deb || sl.From > 0 {
```
Comment (bbfsdev) [inline cmd/search_log.go:392]: Do you skip queries with filters?

File: cmd/search_log.go [hunk 3]
```diff
@@ -354,10 +364,91 @@ func latencyAggregateFn(cmd *cobra.Command, args []string) {
...
+				continue
+			}
+			lang := sl.QueryResult.(map[string]interface{})["language"].(string)
+			if lang == "" {
```
Comment (bbfsdev) [inline cmd/search_log.go:396]: What that part do?

File: cmd/search_log.go [hunk 4]
```diff
@@ -54,6 +56,12 @@ var latencyAggregateCmd = &cobra.Command{
 	Run:   latencyAggregateFn,
 }
 
+var queriesAggregateCmd = &cobra.Command{
+	Use:   "queries_aggregate",
+	Short: "Get most popular queries from ElasticSearch.",
```
Comment (bbfsdev) [inline cmd/search_log.go:61]: Why popular? and not random?

File: cmd/search_log.go [hunk 5]
```diff
@@ -354,10 +364,91 @@ func latencyAggregateFn(cmd *cobra.Command, args []string) {
...
+		}
+	}
+	records := [][]string{}
+	for _, lang := range languages {
+		tcs := []TermAndCount{}
+		for k, v := range queryCountByLang[lang] {
+			tcs = append(tcs, TermAndCount{Term: k, Count: v})
+		}
+		if len(tcs) < RESULTS_FOR_LANGUAGE {
+			log.Errorf("Amount of found terms (%d) for language '%s' is smaller than RESULTS_FOR_LANGUAGE const (%d).", len(tcs), lang, RESULTS_FOR_LANGUAGE)
```
Comment (bbfsdev) [inline cmd/search_log.go:421]: What that means?

File: cmd/search_log.go [hunk 6]
```diff
@@ -354,10 +364,91 @@ func latencyAggregateFn(cmd *cobra.Command, args []string) {
...
+			return
+		}
+		sort.Slice(tcs, func(i, j int) bool {
+			return tcs[i].Count > tcs[j].Count
```
Comment (bbfsdev) [inline cmd/search_log.go:425]: Is this the count of queries with the same text?

File: cmd/search_log.go (hunk 4 above)
Comment (YuriHechter) [reply cmd/search_log.go:61]: Text updated: Get 1000 most popular queries and 1000 random queries for each language from ElasticSearch.

File: cmd/search_log.go (hunk 1 above)
Comment (YuriHechter) [reply cmd/search_log.go:388]: GetLattestQueries bring all search queries until the given date (gteStr param).

Comment (YuriHechter) [reply cmd/search_log.go:388]: Tested it locally with small amount of data

File: cmd/search_log.go (hunk 2 above)
Comment (YuriHechter) [reply cmd/search_log.go:392]: No.

File: cmd/search_log.go (hunk 3 above)
Comment (YuriHechter) [reply cmd/search_log.go:396]: "Lang" has no value when we combine the results of all languages (today this happens with Spanish UI)

File: cmd/search_log.go (hunk 5 above)
Comment (YuriHechter) [reply cmd/search_log.go:421]: If total amount of results for specific language is less than 1000

File: cmd/search_log.go (hunk 6 above)
Comment (YuriHechter) [reply cmd/search_log.go:425]: Yes. Same search term.

**Pull request merged**